File: TestBasedGameC#/TestBasedGameCS/Message.cs
namespace TextBasedAIGame;

public class Message
{
    public string Role { get; set; }
    public string Content { get; set; }

    public Message(string role, string content)
    {
        Role = role;
        Content = content;
    }

    public override string ToString()
    {
        return $"role: {Role}, message: {Content}|";
    }
}

File: TextBasedAiGameJava/src/main/java/main/Message.java
package main;

public class Message {

    private String role;
    private String message;

    public Message(String role, String message) {
        this.role = role;
        this.message = message;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Override
    public String toString() {
        return "role: " + role + ", message: " + message + "|";
    }
}
